pipelines/controller/src/main/java/com/google/fhir/analytics/DwhFilesManager.java

```diff
@@ -364,11 +370,15 @@ Set<ResourceId> getAllChildDirectories(String baseDir) throws IOException {
   }
 
   private int getLastIndexOfSlash(String dwhRootPrefix) {
-    String scheme = parseScheme(dwhRootPrefix);
+    String scheme = DwhFiles.parseScheme(dwhRootPrefix);
     int index = -1;
     switch (scheme) {
       case DEFAULT_SCHEME:
-        index = dwhRootPrefix.lastIndexOf("/");
+        if (isSchemeWindows(dwhRootPrefix)) {
+          index = getLastIndexOfSlashInWindowsSpec(dwhRootPrefix);
+        } else {
+          index = dwhRootPrefix.lastIndexOf(File.separator);
```

Comment:
Why can't we use this in both Windows and Unix cases? I mean why do we need `getLastIndexOfSlashInWindowsSpec`? I am guessing this is because how `getLastIndexOfSlashInWindowsSpec` is used elsewhere, e.g., in `getBaseDir` and `getPrefix` where we have a condition for `index <= 0`. If that's the case I suggest we rename this `getLastIndexOfSlash` to something more meaningful (which throws exceptions) and do all required error-checking is done here.

Reply:
To clarify a little more about this and several other comments I have made in my review: It would be great if we have _no_ Windows specific code in our own repo; instead rely on libraries to handle differences for us, e.g., `java.io*` or `beam.sdk.io*`, etc.

Reply:
Have removed the condition for Windows platform now. This condition for Windows was added so that there is a mandatory base directory in case of absolute path. This condition has been relaxed now for Windows (its mandatory for Unix though). Have updated the comments in the `pipeline/config/application.yaml` file accordingly.